app/main.py

```diff
@@ -1,2 +1,148 @@
+from __future__ import annotations
+
+from dataclasses import dataclass
+from typing import Any, Hashable
+
+
 class Dictionary:
-    pass
+    _DEFAULT_INITIAL_CAPACITY = 8
+    _DEFAULT_LOAD_FACTOR = 2 / 3
+    _DEFAULT_CAPACITY_MULTIPLIER = 2
+
+    @dataclass
+    class Node:
+        key: Hashable
+        value: Any
```

Comment:
In my implementation, the `Node.hash` isn't really used throughout the code. The hash is always calculated inside the `_calculate_index` method depending on the `key`, and it is unknown if this is the key of existing Node with calculated hash or the new one. I decided that it's redundant to store it as an object attribute. But ok, I added it, so the requirements of the condition are met.

Reply:
I would store the hash in a Node to add an extra check that both key and hash are matching when we are going through the indexes of hash table. Just in case, to be 100% sure that we get correct value. It is true though that you might not need to check hash in your implementation